core/storage/question/gae_models.py

```diff
@@ -316,25 +316,58 @@ def get_question_skill_links_by_skill_ids(
         ) * question_count
 
         if not start_cursor == '':
-            cursor = datastore_query.Cursor(urlsafe=start_cursor)
-            question_skill_link_models, next_cursor, more = cls.query(
-                cls.skill_id.IN(skill_ids)
-                # Order by cls.key is needed alongside cls.last_updated so as to
-                # resolve conflicts, if any.
-                # Reference SO link: https://stackoverflow.com/q/12449197
-            ).order(-cls.last_updated, cls.key).fetch_page(
-                question_skill_count,
-                start_cursor=cursor
-            )
+            direction = start_cursor[-1]
```

Comment:
I don't understand the cursor logic and why you're doing indexing with -13, -1, etc. Could you please explain this? It is quite opaque.

Reply:
I am attaching the random_id and direction of fetch to the cursor so that I can create exactly same query in the subsequent calls. The last char is the direction i.e 0 => '>' and 1 => '<=' and next last 12 char is the random id used to filter it. These details are required for subsequent queries otherwise the cursor can't be used.

Reply:
Even I am not completely getting the logic behind this, it might be worthwhile writing a doc explaining your logic. One more general question, I believe the number of pretest questions would be at the most 3-4, so why do we even need to fetch questions by page? Why can't we just pick up all questions and send it in one shot (like how we do it for practice sessions and review tests)? @aks681 @seanlip

Reply:
@rohitkatlaa Can you explain this logic before we go further in the review? It would be great to write up a small doc and give it for review to @aks681 @seanlip and I

Reply:
@vinitamurthi @seanlip @aks681  I have updated the original [doc](https://docs.google.com/document/d/1IUhXcmEps7bJwFvPPBRxXYZ8tpl_zOoETKzQA_Q4vSw/edit?usp=sharing) design explaining the logic used. PTAL.